plugins/BEdita/Core/src/Model/Action/CloneAction.php

```diff
@@ -0,0 +1,251 @@
+    {
+        $this->Table = $this->getConfig('table');
+        $this->authorId = empty(LoggedUser::id()) ? LoggedUser::getUserAdmin()['id'] : LoggedUser::id();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(array $data = [])
+    {
+        $sourceId = (int)Hash::get($data, 'id');
+        $include = (array)Hash::get($data, 'data._meta.include');
+        $attributes = array_filter($data['data'], function ($key) {
+            return !in_array($key, ['_meta']);
+        }, ARRAY_FILTER_USE_KEY);
+        $entity = null;
+        $this->Table->getConnection()->transactional(function () use (&$entity, $sourceId, $attributes, $include) {
+            $objectType = $this->Table->objectType();
+            $options = $objectType->hasAssoc('Streams') ? ['contain' => ['Streams']] : [];
+            $source = $this->Table->get($sourceId, $options);
+            $entity = $this->cloneEntity($source, $attributes);
+            if (!empty($source->get('streams'))) {
+                $this->cloneStreams($source->get('streams'), $entity);
+            }
+            if (in_array('relationships', $include)) {
+                $this->cloneRelationships($sourceId, $entity->id);
+            }
+            if (in_array('translations', $include)) {
+                $this->cloneTranslations($sourceId, $entity->id);
+            }
+        });
+
+        return $entity;
+    }
+
+    /**
+     * Clone entity
+     *
+     * @param \BEdita\Core\Model\Entity\ObjectEntity $sourceEntity Source object
+     * @param array $attributes Attributes to set
+     * @return \Cake\Datasource\EntityInterface
+     */
+    public function cloneEntity(ObjectEntity $sourceEntity, array $attributes): EntityInterface
+    {
+        $schema = $this->Table->getSchema();
+        $reset = Schema::getPrimaryFields($schema) + ['created', 'modified'];
+        $unique = Schema::getUniqueFields($schema);
+        $nullable = Schema::getNullableFields($schema);
+        $schemaInfo = compact('reset', 'unique', 'nullable', 'attributes');
+        /** @var \BEdita\Core\Model\Entity\ObjectEntity $entity */
+        $entity = $this->Table->newEmptyEntity();
+        $entityAttributes = $sourceEntity->getVisible();
+        foreach ($entityAttributes as $field) {
+            $this->setEntityField($schemaInfo, $sourceEntity, $entity, $field);
+        }
+        $entity->set('created_by', $this->authorId);
+        $entity->set('modified_by', $this->authorId);
+
+        return $this->Table->saveOrFail($entity);
+    }
+
+    /**
+     * Set entity field
+     *
+     * @param array $schemaInfo Schema information
+     * @param \BEdita\Core\Model\Entity\ObjectEntity $sourceEntity Source object
+     * @param \BEdita\Core\Model\Entity\ObjectEntity $entity Destination object
+     * @param string $field Field name
+     * @return mixed
+     */
+    protected function setEntityField(array $schemaInfo, ObjectEntity $sourceEntity, ObjectEntity &$entity, string $field)
```

Comment:
```suggestion
    protected function setEntityField(array $schemaInfo, ObjectEntity $sourceEntity, ObjectEntity $entity, string $field)
```

Objects are always passed by reference.